packages/shared/src/components/modals/report/ReportModal.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement, ReactNode, useState } from 'react';
+import { Radio, RadioOption } from '../../fields/Radio';
+import { Button } from '../../buttons/Button';
+import { Modal, ModalProps } from '../common/Modal';
+import { Justify } from '../../utilities';
+import { ReportReason } from '../../../graphql/posts';
+import { ReportCommentReason } from '../../../graphql/comments';
+
+interface Props extends ModalProps {
+  onReport(
+    e: React.MouseEvent,
+    reason: ReportReason | ReportCommentReason,
+    text: string,
+  ): void;
+  reasons: RadioOption[] | ((reason: string) => RadioOption[]);
+  heading: string;
+  title?: string;
+  footer?: ReactNode;
+  disabled?: boolean;
+}
+
+export const OTHER_KEY = 'OTHER';
+
+export function ReportModal({
+  onReport,
+  reasons,
+  heading,
+  title,
+  footer,
+  disabled,
+  ...props
+}: Props): ReactElement {
+  const [reason, setReason] = useState(null);
+  const [note, setNote] = useState<string>();
+
+  return (
+    <Modal isOpen kind={Modal.Kind.FixedCenter} {...props}>
+      <Modal.Header title={heading} />
+      <Modal.Body className="py-5">
+        {title && (
+          <p className="mb-6 text-theme-label-tertiary typo-callout">{title}</p>
+        )}
+        <Radio
+          name="report_reason"
+          options={typeof reasons === 'function' ? reasons(reason) : reasons}
+          value={reason}
+          onChange={setReason}
+        />
+        {reason === OTHER_KEY && (
```

Comment:
For post modal the textarea is always shown but optional to fill in. It is required when `reason === OTHER_KEY`. Not sure if the spec changed but this should remain as it is now.

Reply:
It should be same logic for comment report modal.

Reply:
Oh, wow. Thank you for the link to the discussion. We should probably ask the design team to update the Sketch too in order to avoid confusion.

Reply:
So this should always render correct?
Not sure why it's marked as resolved.

Reply:
@sshanzel @capJavert, Could you please confirm, the way i read the above discussion was the textarea only showed when the Other was selected. But could you please confirm

Reply:
It should always show.

Reply:
Apologies i will then fix this then